Content.Server/Administration/BanList/BanListEui.cs

```diff
@@ -65,21 +65,40 @@ private async Task LoadBans(NetUserId userId)
                 unban = new SharedServerUnban(unbanningAdmin, ban.Unban.UnbanTime.UtcDateTime);
             }
 
-            Bans.Add(new SharedServerBan(
-                ban.Id,
-                ban.UserId,
-                ban.Address is { } address
-                    ? (address.address.ToString(), address.cidrMask)
-                    : null,
-                ban.HWId == null ? null : Convert.ToBase64String(ban.HWId.Value.AsSpan()),
-                ban.BanTime.UtcDateTime,
-                ban.ExpirationTime?.UtcDateTime,
-                ban.Reason,
-                ban.BanningAdmin == null
-                    ? null
-                    : (await _playerLocator.LookupIdAsync(ban.BanningAdmin.Value))?.Username,
-                unban
-            ));
+            if (!_admins.HasAdminFlag(Player, AdminFlags.Pii))
+            {
+                Bans.Add(new SharedServerBan(
+                    ban.Id,
+                    ban.UserId,
+                    ("*Hidden*",0),
+                    "*Hidden*",
+                    ban.BanTime.UtcDateTime,
+                    ban.ExpirationTime?.UtcDateTime,
+                    ban.Reason,
+                    ban.BanningAdmin == null
+                        ? null
+                        : (await _playerLocator.LookupIdAsync(ban.BanningAdmin.Value))?.Username,
+                    unban
+                ));
+            }
+            else
+            {
+                Bans.Add(new SharedServerBan(
+                    ban.Id,
+                    ban.UserId,
+                    ban.Address is { } address
+                        ? (address.address.ToString(), address.cidrMask)
+                        : null,
+                    ban.HWId == null ? null : Convert.ToBase64String(ban.HWId.Value.AsSpan()),
+                    ban.BanTime.UtcDateTime,
+                    ban.ExpirationTime?.UtcDateTime,
+                    ban.Reason,
+                    ban.BanningAdmin == null
+                        ? null
+                        : (await _playerLocator.LookupIdAsync(ban.BanningAdmin.Value))?.Username,
+                    unban
+                ));
+            }
```

Comment:
I suggest combining these and instead making an `address` and `hwid` variable that gets set inside the Pii flag check if.

Reply:
done